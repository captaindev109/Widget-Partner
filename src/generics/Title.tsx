import styled, { DefaultTheme } from 'styled-components';

export interface TitleProps {
  children?: string;
  variant?: 'primary' | 'secondary';
}

export interface VariantProps {
  variant?: string;
  theme?: DefaultTheme;
}

function fontSize({ variant }: VariantProps) {
  switch (variant) {
    case 'secondary':
      return `font-size: 18px`;
    default:
      return `font-size: 24px`;
  }
}

function lineHeight({ variant }: VariantProps) {
  switch (variant) {
    case 'secondary':
      return `line-height: 26px`;
    default:
      return `line-height: 32px`;
  }
}

const Title = styled.div<TitleProps>`
  color: #121212;
  font-family: Favorit;
  ${fontSize};
  ${lineHeight};
`;

Title.defaultProps = {
  variant: 'primary',
};

export default Title;
