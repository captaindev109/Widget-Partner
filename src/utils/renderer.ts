import React from 'react';
import { ConfigProps, Index, EachComponentObjectChildren, BasicAny } from '@/types/types';
import { UIComponents } from './components';

const recursivelyCreateReactComponents = (
  componentToCreate: BasicAny,
  configuration: ConfigProps,
  children: BasicAny
) => {
  if (typeof componentToCreate !== 'undefined') {
    return React.createElement(
      componentToCreate,
      configuration,
      children &&
        (typeof children === 'string'
          ? children
          : children.map((component: EachComponentObjectChildren, index: Index) => viewRenderer(component, index)))
    );
  }
  null;
};

const viewRenderer: BasicAny = (config: ConfigProps, index: Index) => {
  const children = config.children;
  const newConfig = { ...config, key: index };
  const LookForComponentInUILibrary = UIComponents[config.component];
  return recursivelyCreateReactComponents(LookForComponentInUILibrary, newConfig, children);
};

export default viewRenderer;
