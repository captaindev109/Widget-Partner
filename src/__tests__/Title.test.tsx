import { render, RenderResult } from '@testing-library/react';
import queries from '@testing-library/dom/types/queries';
import renderer from 'react-test-renderer';
import Title from '../generics/Title';

const children = 'Welcome to Hedvig!';

describe('Test cases of the Title component', () => {
  let renderedComponent: RenderResult<typeof queries, HTMLElement>;

  beforeEach(() => {
    renderedComponent = render(<Title variant="primary">{children}</Title>);
  });

  test('to see if the component render without crashing', () => {
    render(<Title />);
  });

  test('to see if the title src is correct', () => {
    const { container } = renderedComponent;
    expect(container.textContent).toBe(children);
  });

  it('Snapshot test should pass', () => {
    const snap = renderer.create(<Title />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
