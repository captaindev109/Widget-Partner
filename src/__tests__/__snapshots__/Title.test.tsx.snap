// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases of the Title component Snapshot test should pass 1`] = `
<div
  className="Title-sc-spq86-0 dDCGeH"
/>
`;
